src/test/java/com/glencoesoftware/omero/zarr/ZarrPixelBufferTest.java

```diff
@@ -503,8 +503,40 @@ public void testGetTileLargerThanImage()
         }
     }
 
-    @Test
-    public void testTileExceedsMax() throws IOException, InvalidRangeException {
+    @Test(expected = IllegalArgumentException.class)
+    public void testTileIntegerOverflow()
+            throws IOException, InvalidRangeException {
+        int sizeT = 1;
+        int sizeC = 3;
+        int sizeZ = 1;
+        int sizeY = 1;
+        int sizeX = 1;
+        int resolutions = 1;
+        Pixels pixels = new Pixels(
+                null, null, sizeX, sizeY, sizeZ, sizeC, sizeT, "", null);
+        Path output = writeTestZarr(
+                sizeT, sizeC, sizeZ, sizeY, sizeX, "uint16",
+                resolutions);
+
+        // Hack the .zarray so we can appear as though we have more data than
+        // we actually have written above.
```

Comment:
I assume the rationale if that is we want to generate an OME-Zarr of a given dimensions without actually writing the underlying test data (which increases the test time for no value).

@melissalinkert this makes me wonder whether we could introduce an option in `bioformats2raw`  to write sparse data to quickly generate such datasets.

Reply:
> I assume the rationale if that is we want to generate an OME-Zarr of a given dimensions without actually writing the underlying test data (which increases the test time for no value).

Correct. Creating an array large enough to not end up with earlier out of bounds checks can take minutes.